sbin/firehol

```diff
@@ -25,31 +25,28 @@
 #       See the file COPYING for details.
 #
 
-VERSION='$Id$'
-PROGRAM_FILE="${0}"
-PROGRAM_DIR="${0%/*}"
-if [ "$PROGRAM_DIR" = "$0" ]; then PROGRAM_DIR="."; fi
+PROGRAM_FILE="$(/bin/readlink $0)"
+PROGRAM_FILE="${PROGRAM_FILE:-$0}"
```

Comment:
I have busybox readlink -f working, strange.

I do not think you should use full path.